### PR TITLE
auto-generate Validate functions

### PR DESCRIPTION
**Pull request by jasonbornsteinMOOV**

### CODE REVIEW

**Comment by adamdecaf** [comment]
We should be able to get back above 60% coverage by testing the validation helpers added in this PR.

File: pkg/rtp/restrictions.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	for indx := range enumeration {
+		if enumeration[indx] == value {
+			return nil
+		}
+	}
+	return fmt.Errorf("%s fails enumeration validation", value)
+}
+
+func ValidateLength(value string, length int) error {
+	if len(value) != length {
```
[the remark this answers is not in the document]
Comment (jasonbornsteinMOOV) [reply pkg/rtp/restrictions.go:44]: I did a simple test where I inserted `&amp;` into a name in the XML and it counted the characters correctly using both `len(value)` and `utf8.RuneCountInString(value)`

[the remark this answers is not in the document]
Comment (adamdecaf) [reply pkg/rtp/restrictions.go:44]: Can we test some non-english characters like the PDF mentions?

[the remark this answers is not in the document]
Comment (jasonbornsteinMOOV) [reply pkg/rtp/restrictions.go:44]: Interesting. I added another UTF-8 non-standard character `€` and got the results you were expecting to see. this character is counted as 1 from `RuneCountInString` and 3 from `len(value)`.

[the remark this answers is not in the document]
Comment (adamdecaf) [reply pkg/rtp/restrictions.go:44]: https://go.dev/blog/strings is a good read

**Review by adamdecaf — APPROVED** [review]

**Pull request merged**